src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs

```diff
@@ -412,6 +412,7 @@ private void ParseArgs()
             Parser parser = new Parser(parseConfig);
             _parseResult = parser.Parse(argsWithCommand.ToArray());
             _templateParamCanonicalToVariantMap = null;
+            _templateParamVariantToCanonicalMap = null;
```

Comment:
minor:
The existing pattern is to null out maps & this change merely abided to such pattern.  Question about the choice of `IReadOnlyDictionary` prevents "reset" via calling Clear on mutable Dictionaries.  `_templateParamVariantToCanonicalMap` is...
- a private instance member
- Initialized in a method that returns `IReadOnlyDictionary`
- the state of `_templateParamVariantToCanonicalMap` seems to only be needed to check if it is null

I don't see compelling reasons why this map must be hid behind `IReadOnlyDictionary`.

Reply:
in this case, why `IReadOnlyDictionary` & not https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutabledictionary-2?view=net-5.0 ?

Reply:
This class and its data structures are not perfect. Since we are planning to move new CLI library potentially this class will die very soon, therefore I don't think it worth to refactor it now.